### PR TITLE
Move `SkoptSampler`

### PR DESCRIPTION
**Pull request by Alnusjaponica**
## Motivation
Progress https://github.com/optuna/optuna/issues/4484

## Description of the changes
No tests have transferred since SkopSampler is highly unrecommended and not tested in optuna/optuna

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,6 +23,7 @@ Optuna-Integration API reference is [here](https://optuna-integration.readthedoc
 * [ChainerMN](https://optuna-integration.readthedocs.io/en/stable/reference/index.html#chainermn)  ([example](https://github.com/optuna/optuna-examples/tree/main/chainer/chainermn_simple.py))
 * [Keras](https://optuna-integration.readthedocs.io/en/stable/reference/index.html#keras)  ([example](https://github.com/optuna/optuna-examples/tree/main/keras))
 * [MXNet](https://optuna-integration.readthedocs.io/en/stable/reference/index.html#mxnet)  ([example](https://github.com/optuna/optuna-examples/tree/main/mxnet))
+* [scikit-optimize](https://optuna-integration.readthedocs.io/en/stable/reference/index.html#skopt)  ([example](
```
Comment (y0z) [inline README.md:26]: It looks like the link is broken.
image

File: docs/source/reference/index.rst [hunk 2]
```diff
@@ -64,6 +64,15 @@ MXNet
...
    optuna.integration.MXNetPruningCallback
 
+scikit-optimize
+---------------
+
+.. autosummary::
+   :toctree: generated/
+   :nosignatures:
+
+   optuna.integration.SkoptSampler
```
Comment (y0z) [inline docs/source/reference/index.rst:74]: The documentation should be generated from `optuna_integration`, not `optuna.integration` .
```suggestion
   optuna_integration.SkoptSampler
```

File: optuna_integration/skopt.py [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+            Note that the parameters of the first trial in a study are always sampled
+            via an independent sampler, so no warning messages are emitted in this case.
+
+        skopt_kwargs:
+            Keyword arguments passed to the constructor of
+            `skopt.Optimizer <https://scikit-optimize.github.io/#skopt.Optimizer>`_
+            class.
+
+            Note that ``dimensions`` argument in ``skopt_kwargs`` will be ignored
+            because it is added by :class:`~optuna.integration.SkoptSampler` automatically.
```
Comment (y0z) [inline optuna_integration/skopt.py:69]:
```suggestion
            because it is added by :class:`~optuna_integration.SkoptSampler` automatically.
```

**Review by y0z — CHANGES_REQUESTED** [review]
I leave some comments.

In addition to the above comments,
- please add `SkoptSampler` to [`__init__.py`](https://github.com/optuna/optuna-integration/blob/main/optuna_integration/__init__.py) like other integrations,
- because this PR does not include any tests, could you provide minimal code like the one below to verify the `SkoptSampler` works as intended for reviewers?

```python
import optuna
from optuna_integration import SkoptSampler

study = optuna.create_study(sampler=SkoptSampler())

...

study.optimize(...
```

File: docs/source/reference/index.rst (hunk 2 above)
Comment (Alnusjaponica) [reply docs/source/reference/index.rst:74]: I wonder why `SkoptSampler` is different from other modules, e.g., `optuna.integration.MXNetPruningCallback` in the 65th line.

File: README.md (hunk 1 above)
Comment (Alnusjaponica) [reply README.md]: Thanks for the comment. I removed it.

File: docs/source/reference/index.rst (hunk 2 above)
Comment (y0z) [reply docs/source/reference/index.rst:74]: I think [the documentation in the current main branch](https://github.com/optuna/optuna-integration/blob/main/docs/source/reference/index.rst?plain=1) is generated from `optuna_integration`.
Is your branch up-to-date?

Comment (Alnusjaponica) [reply docs/source/reference/index.rst]: You're right. I updated this branch and fixed it.

File: optuna_integration/skopt.py [hunk 4]
```diff
@@ -0,0 +1,345 @@
+import copy
+import random
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Sequence
+from typing import Tuple
```
[the remark this answers is not in the document]
Comment (Alnusjaponica) [reply optuna_integration/skopt.py]: Thanks for the comment. I updated.

**Review by nabenabe0928 — APPROVED** [review]
LGTM

**Comment by y0z** [comment]
I think these comments have not been resolved.
> perhaps, scikit-optimize needs to be added to pyproject.toml (cf. [here](https://optuna.readthedocs.io/en/stable/reference/integration.html#dependencies-of-each-integration)),
> because this PR does not include any tests, could you provide minimal code like the one below to verify the SkoptSampler works as intended for reviewers?

I've made another PR (https://github.com/optuna/optuna-integration/pull/78) as a follow-up task.
Note that our CI cannot detect dependency errors because no tests are prepared for `SkoptSampler`.

**Pull request merged**